sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -292,14 +296,17 @@ trait CreateNamedStructLike extends Expression {
   }
 
   override def checkInputDataTypes(): TypeCheckResult = {
-    if (children.size % 2 != 0) {
+    if (children.length < 1) {
+      TypeCheckResult.TypeCheckFailure(
+        s"input to function $prettyName requires at least one argument")
```

Comment:
This is not related to what this PR claims to do. What's the reason behind this change?

Reply:
This is a behavior change and caused a problem in https://github.com/apache/spark/pull/22373

Reply:
sorry, but I don't remember correctly.
I looked over this pr again and I also think the modification is not related to this pr.